### DISCUSSION TITLE
Alfred 5 compatibility

### DISCUSSION THREAD

**Discussion by zjn0505** [xilopaint/alfred-pdf-tools#23] [General]
Hi, 
I am using this workflow lib to build my workflows.

Today I tried it on Alfred 5 and it failed because of pickle protocol.

`ValueError: unsupported pickle protocol: 5`

The workflow is using python 3.7.9 from /usr/local/bin/python3

It looks that I have to change pickle protocol to 4 when writing to cache

https://github.com/xilopaint/alfred-pdf-tools/blob/main/src/workflow/workflow.py#L605

**Comment by xilopaint** [comment]
Pickle protocol 5 was introduced by [PEP 574](https://peps.python.org/pep-0574/) to work with Python 3.8, that's why it doesn't work in your 3.7.9 environment. It's not a compatibility issue with Alfred 5. Also, the Python version installed by CLT is 3.8.9 and I do not intend to support any earlier version.

It looks like you're trying to dump to a dataframe originally created with Python 3.8+ and Pickle protocol 5 which is unsupported in Python 3.7.9.
